### PR TITLE
Add kgs reference docs

### PR DESCRIPTION
**Pull request by marians**
This adds the `kubectl gs` reference section.

### CODE REVIEW

File: src/content/reference/kubectl-gs/get-clusters.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    - `UPDATING`: The cluster is currently being updated, e. g. during an upgrade.
+    - `UPDATED`: The cluster update is finished.
+    - `DELETING`: The cluster is being deleted.
+- `RELEASE`: Release version of the cluster.
+- `ORGANIZATION`: Organization owning the cluster.
+- `DESCRIPTION`: User friendly description for the cluster.
+
+## Flags {#flags}
+
+Here we document the flags that have a particular meaning for the `get clusters` command. Use `kgs get clusters -h` for a full list.
```
Comment (axbarsan) [inline src/content/reference/kubectl-gs/get-clusters.md:62]: I believe we should provide the complete flag names

```suggestion
Here we document the flags that have a particular meaning for the `get clusters` command. Use `kgs get clusters --help` for a full list.
```

File: src/content/reference/kubectl-gs/installation.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+```nohighlight
+kgs
+```
+
+You should see information regarding the commands available.
+
+To upgrade to the latest version of the plug-in, use this command:
+
+```nohighlight
+krew upgrade gs
```
Comment (axbarsan) [inline src/content/reference/kubectl-gs/installation.md:44]:
```suggestion
kubectl krew upgrade gs
```

**Review by pipo02mix — APPROVED** [review]
nice

File: src/content/reference/kubectl-gs/template-appcatalog.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+The command to execute is `kubectl gs template appcatalog`.
+
+It supports the following flags:
+
+- `--name` - Catalog name.
+- `--description` - Description of the purpose of the catalog.
+- `--url` - URL where the helm repository lives.
+- `--owner` - organization, owning tenant cluster. Must be configured with existing organization in installation.
+- `--logo` - tenant cluster AWS region. Must be configured with installation region.
```
Comment (axbarsan) [inline src/content/reference/kubectl-gs/template-appcatalog.md:27]:
```suggestion
- `--logo` - URL of the catalog logo image.
```

File: src/content/reference/kubectl-gs/template-cluster.md [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+
+**Note:** The CRs generated won't trigger the creation of any worker nodes. Please see the [template nodepool](/reference/kubectl-gs/template-nodepool/) for instructions on how to create worker node pools.
+
+## Usage
+
+The command to execute is `kubectl gs template cluster`.
+
+It supports the following flags:
+
+- `--provider` - The infrastructure provider (either `aws` or `azure`)
```
Comment (axbarsan) [inline src/content/reference/kubectl-gs/template-cluster.md:38]: Consistency with the `template nodepool` command

```suggestion
- `--provider` - The infrastructure provider (either *aws* or *azure*)
```

**Review by axbarsan — COMMENTED** [review]
Thank you for this. Just a few comments

File: src/content/reference/kubectl-gs/template-appcatalog.md (hunk 3 above)
Comment (marians) [reply src/content/reference/kubectl-gs/template-appcatalog.md:27]: Good catch, thanks!

File: src/content/reference/kubectl-gs/template-cluster.md (hunk 4 above)
Comment (marians) [reply src/content/reference/kubectl-gs/template-cluster.md:38]: We should use backticks for literals. I'll align the other command then.

**Pull request merged**